app/assets/javascripts/angular/controllers/dashboard_ctrl.js

```diff
@@ -30,6 +30,31 @@ angular.module("Prometheus.controllers")
     $scope.$broadcast('redrawGraphs');
   }
 
+  $scope.generateStaticDashboard = function(event) {
+    // TODO: When the input with the static dashboard link is visible,
+    // should clicking the "generate" button close the input, or
+    // generate a NEW static link?
+    $scope.generatingPermalink = true;
```

Comment:
perhaps add a line before this that just returns when the link is already currently being generated?